tests/modules/test_completion.py

```diff
@@ -0,0 +1,76 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from nf_core.modules._completion import autocomplete_modules
+
+
+class DummyParam:
+    # Minimal mock object for Click parameter (not used in the function)
+    pass
+
+
+class DummyCtx:
+    def __init__(self, obj=None):
+        self.obj = obj
+
+
+@patch("nf_core.modules._completion.CompletionItem")
+@patch("nf_core.modules._completion.ModuleList")
+def test_autocomplete_modules_mocked(mock_module_list_class, mock_completion_item_class):
+    # Setup mock for module list
+    mock_instance = mock_module_list_class.return_value
+    mock_instance.modules_repo.get_avail_components.return_value = ["fastqc", "bcftools/call", "bcftools/index"]
+
+    # Setup mock for CompletionItem
+    def mock_completion(value):
+        mock_item = MagicMock()
+        mock_item.value = value
+        return mock_item
+
+    mock_completion_item_class.side_effect = mock_completion
+
+    ctx = DummyCtx()
+    param = DummyParam()
+    completions = autocomplete_modules(ctx, param, "bcf")
+
+    values = [c.value for c in completions]
+    assert "bcftools/call" in values
+    assert "fastqc" not in values
+
+
+@patch("nf_core.modules._completion.ModuleList")
+def test_autocomplete_modules_with_ctx_obj(mock_module_list_class):
+    # Setup mock return value
+    mock_instance = mock_module_list_class.return_value
+    mock_instance.modules_repo.get_avail_components.return_value = ["custommodule/a", "custommodule/b", "othermodule/x"]
```

Comment:
can this test ever fail? looks like we mock here the whole logic, so nothing can really happen, no? 
Is there a need to mock the available compolnents instead of using the actual ones?

Reply:
If I do so running it takes more than a minute.

Reply:
Done !
This takes now 3 seconds only 😄